packages/tpetra/core/ext/TpetraExt_MatrixMatrix_decl.hpp

```diff
@@ -648,6 +542,105 @@ void setMaxNumEntriesPerRow(
 template<class CrsMatrixType>
 size_t C_estimate_nnz(CrsMatrixType & A, CrsMatrixType &B);
 
+template<typename Scalar, typename LocalOrdinal, typename GlobalOrdinal, typename Node>
+struct AddKernels
+{
+  typedef Tpetra::CrsMatrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> crs_matrix_type;
```

Comment:
This is a copy-and-paste so don't worry for now, but prefer C++11 alias syntax `using new_type = old_type;` to `typedef new_type old_type;` for future code.